File: programmers/pg_lv2_이진변환반복하기.js
function solution(s) {
  let changeCount = 0;
  let zeroCount = 0;

  while (s != 1) {
    let originLeng = s.length; //처음 s 길이

    s = s.replace(/0/g, "");
    let newLeng = s.length; //0을 제거한 뒤의 s 길이

    zeroCount += originLeng - newLeng;

    s = newLeng.toString(2); //s를 이진 변환 함
    changeCount++;
  }

  var answer = [changeCount, zeroCount];
  return answer;
}

File: python/programmers/pg_lv2_이진변환반복하기.py
def solution(s):
    if len(s) == 1: return [0, 0]

    changeCount = 0
    zeroCount = 0
    
    while len(s) != 1:
        newSLen = len(s.replace("0", ""))
        zeroCount += (len(s) - newSLen)
        changeCount += 1
        
        if newSLen != 1: s = str(bin(newSLen)[2:])
        else: return [changeCount, zeroCount]